infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/GeometryService.kt

```diff
@@ -198,9 +198,20 @@ class GeometryService @Autowired constructor(
 
     fun getPlanFile(planId: IntId<GeometryPlan>): InfraModelFile {
         logger.serviceCall("getPlanFile", "planId" to planId)
-        return geometryDao.getPlanFile(planId)
+
+        val fileAndSource = geometryDao.getPlanFile(planId)
+        return if (fileAndSource.source == PAIKANNUSPALVELU) {
+            InfraModelFile(
+                fileNameWithSourcePrefixIfPaikannuspalvelu(fileAndSource.file.name, fileAndSource.source),
+                fileAndSource.file.content
+            )
+        } else fileAndSource.file
     }
 
+    private fun fileNameWithSourcePrefixIfPaikannuspalvelu(originalFileName: FileName, source: PlanSource): FileName =
+        if (source == PAIKANNUSPALVELU) FileName("PAIKANNUSPALVELU_EPÄLUOTETTAVA_$originalFileName")
```

Comment:
Meillä sietäisi olla kyllä joku backarin lokalisointisysteemi ennen kun näitä kovakoodattuja stringejä leviää enempää (nyt csv ja tämä). Tässä casessa olisi tietenkin myös toinen vaihtoehto että annetaan nameprefix queryparamina frontilta. Hiukan hassu tavallaan, mutta lokalisoinnit on frontissa muutenkin ja frontti näyttää jo muutenkin varoituspopupin -> iffittely on jo olemassa. Etuna olisi että backarissa ei tarvitisi tehdä mitään muuta kuin lisätä liite aina tiedostonimeen palautuksessa, ylimääräistä tyyppiä tai service/dao muutoksia ei tarvitsisi lainkaan.

Reply:
Mietin prefixin tuomista frontilta itsekin, mutta sen arpominen kuuluisi mun mielestäni enemmän bäkkärin harteile. Mielestäni tuo fiksu lokalisaatiosysteemi olisi oikeampi ratkaisu.